mmv1/third_party/terraform/provider/provider_test.go.tmpl

```diff
@@ -166,6 +168,115 @@ func TestAccProviderEmptyStrings(t *testing.T) {
 	})
 }
 
+// TestAccProvider_external_credentials_upgrade verifies that credentials and access_token
+// continue to function properly when upgrading to a version with external_credentials
+func TestAccProvider_external_credentials_upgrade(t *testing.T) {
```

Comment:
Thanks for this inclusion, I like having an automated upgrade test. I will note that this is missing a ADC or impersonation upgrade case, but those feel safe as the new logic is gated behind a very simple nil check: https://github.com/GoogleCloudPlatform/magic-modules/pull/12714/files#diff-2f2e11cd40a91d9e9b280c00ad695d0d71e8d43488a3026e226fc98a79b20776R306